app/validators/previously_published_validator.rb

```diff
@@ -0,0 +1,14 @@
+class PreviouslyPublishedValidator < ActiveModel::Validator
+  def validate(record)
+    if record.previously_published.nil?
+      record.errors[:base] << "You must specify whether the document has been published before"
+      record.has_previously_published_error = true
+    elsif record.previously_published
+      if record.first_published_at.blank?
+        record.errors.add(:first_published_at, "can't be blank")
+      elsif record.first_published_at > Time.zone.now
```

Comment:
so if I understand correct this means that:

all records that have been previously published cannot have a first_published_at in the future and thus the inverse that any record that has not been previously published can have a first_published_at in the future?

Would the bug not still occur when someone tries to associate with a government and has a first_published_at date in the future but the document is not previously published?

Reply:
Both good points, I've attempted to address in the [following commit](https://github.com/alphagov/whitehall/pull/4076/commits/3adb1d2b560e2d77bf560b7b22cef10793a296be).
I think the UI behaviour would prevent the manual creation of a non-previously published doc with a future `first_published_at` date which is reason enough to make sure the validator covers this scenario. 
I've also falsified the `has_previously_published_error` var before validating.